src/api/managers/roles/GuildRolesManager.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  EntitiesCacheManager,
+  RawRoleCreateData,
+  Role,
+  RoleCreateData
+} from '@src/api'
+import { EntitiesManager } from '@src/api/managers/EntitiesManager'
+import { RestEligibleDiscordApplication } from '@src/core'
+import { DiscordooError, resolveGuildId } from '@src/utils'
+import { GuildRolesManagerData } from '@src/api/managers/roles/GuildRolesManagerData'
+
+export class GuildRolesManager extends EntitiesManager {
+  public cache: EntitiesCacheManager<Role>
+  public guildId: string
+
+  constructor(app: RestEligibleDiscordApplication, data: GuildRolesManagerData) {
+    super(app)
+
+    const id = resolveGuildId(data.guild)
+    if (!id) throw new DiscordooError('GuildRolesManager', 'Cannot operate without guild id provided.')
+    this.guildId = id
+
+    this.cache = new EntitiesCacheManager<Role>(this.app, {
+      keyspace: 'roles',
+      storage: this.guildId,
+      entity: 'Role',
+      policy: 'roles'
+    })
+  }
+
+  async fetch(id: string): Promise<Role | undefined> {
```

Comment:
fetch() should automatically decide whether to use fetchOne or fetchMany. also no fetchMany method is present